scripts/databases/seed.sh

```diff
@@ -3,23 +3,44 @@
 set -euo pipefail
 
 GROUP=$1
+shift
+DATABASE_VERSIONS=$*
 
 function createDatabase() {
-  echo "Downloading database... $1"
+  echo "Downloading database $1 ..."
+
+  mkdir -p "$HOME/Downloads"
   curl -C - "$2" -o "$HOME/Downloads/$1.sql.gz"
-  #curl "$2" -o "$HOME/Downloads/$1.sql.gz"
 
-  echo "Login..."
+  echo "Login ..."
+  rm .access_token_cache # to make sure we're not using an expired token if we're seeding a lot of databases
   source ./auth.sh
 
-  echo "Uploading database...$1"
+  echo "Uploading database $1 ..."
   ./upload.sh "$GROUP" "$HOME/Downloads/$1.sql.gz"
+  echo # empty line to improve output readability
 }
 
-## https://databases.dhis2.org/
-createDatabase "Sierra Leone - 2.39.0" https://databases.dhis2.org/sierra-leone/2.39.0/dhis2-db-sierra-leone.sql.gz
-createDatabase "Sierra Leone - 2.38.1" https://databases.dhis2.org/sierra-leone/2.38.1/dhis2-db-sierra-leone.sql.gz
-createDatabase "Sierra Leone - 2.37.7" https://databases.dhis2.org/sierra-leone/2.37.6/dhis2-db-sierra-leone.sql.gz
-createDatabase "Sierra Leone - 2.36.11" https://databases.dhis2.org/sierra-leone/2.36.0/dhis2-db-sierra-leone.sql.gz
+if [[ -z "$*" ]]; then
+  DATABASE_VERSIONS=("dev")
+
+  # Filter out all supported versions - current + 2 previous, including patch versions (like 2.40, 2.40.0, 2.39, 2.39.0, etc)
+  dhis2_versions=$(curl -fsSL https://releases.dhis2.org/v1/versions/stable.json)
+  latest_version=$(echo "$dhis2_versions" | jq -r '.versions[] | select(.latest == true) | .version')
+  # shellcheck disable=SC2207
+  DATABASE_VERSIONS+=($(
+    echo "$dhis2_versions" |
+    jq -r "[.versions[] |
+    select(.version == $latest_version or .version == ($latest_version - 1) or .version == ($latest_version - 2)) |
+    [.name] + [(.patchVersions[] | select(.hotfix != true) | .name)]] | flatten | .[]"
+  ))
+fi
+
+echo "Seeding the following databases:"
+echo "${DATABASE_VERSIONS[@]}"
+
+for VERSION in "${DATABASE_VERSIONS[@]}"; do
+  createDatabase "Sierra Leone - $VERSION" "https://databases.dhis2.org/sierra-leone/$VERSION/dhis2-db-sierra-leone.sql.gz"
+done
 
-./list.sh | jq '.[].Databases[]? | .ID, .Name, .Url'
+./list.sh | jq '.[].databases[]? | .id, .name, .url'
```

Comment:
I'm wondering if the piping to jq actually should be part of the script? It makes it a lot less flexible.

Another problem is that it lists for all groups regardless of the input so it can be a bit misleading too

Reply:
It's not really needed, yes. I'll go ahead and remove it. 👍